applications/StructuralMechanicsApplication/custom_utilities/constitutive_law_utilities.cpp

```diff
@@ -942,6 +942,257 @@ Matrix ConstitutiveLawUtilities<TVoigtSize>::CalculateDirectPlasticDeformationGr
 /***********************************************************************************/
 /***********************************************************************************/
 
+template<SizeType TVoigtSize>
+void ConstitutiveLawUtilities<TVoigtSize>::CalculateAnisotropicStressMapperMatrix(
+    ConstitutiveLaw::Parameters& rValues,
+    Matrix& rAs,
+    Matrix& rAsInv
+)
+{
+    auto &r_mat_props = rValues.GetMaterialProperties();
+    if (rAs.size1() != VoigtSize || rAs.size2() != VoigtSize) {
+        rAs.resize(VoigtSize, VoigtSize);
+    }
+    noalias(rAs) = ZeroMatrix(VoigtSize, VoigtSize);
+    if (rAsInv.size1() != VoigtSize || rAsInv.size2() != VoigtSize) {
+        rAsInv.resize(VoigtSize, VoigtSize);
+    }
+    noalias(rAsInv) = ZeroMatrix(VoigtSize, VoigtSize);
+    const Vector &r_iso_aniso_yield_ratios = r_mat_props[ISOTROPIC_ANISOTROPIC_YIELD_RATIO];
+    KRATOS_ERROR_IF_NOT(r_iso_aniso_yield_ratios.size() == VoigtSize) << "The length of the ISOTROPIC_ANISOTROPIC_YIELD_RATIO is not correct" << std::endl;
+
+    if (VoigtSize == 6) {
+        rAs(0, 0) = r_iso_aniso_yield_ratios(0);
+        rAs(1, 1) = r_iso_aniso_yield_ratios(1);
+        rAs(2, 2) = r_iso_aniso_yield_ratios(2);
+        rAs(3, 3) = r_iso_aniso_yield_ratios(3);
+        rAs(4, 4) = r_iso_aniso_yield_ratios(4);
+        rAs(5, 5) = r_iso_aniso_yield_ratios(5);
+    } else {
+        rAs(0, 0) = r_iso_aniso_yield_ratios(0);
+        rAs(1, 1) = r_iso_aniso_yield_ratios(1);
+        rAs(2, 2) = r_iso_aniso_yield_ratios(2);
+    }
+    for (IndexType i = 0; i < VoigtSize; ++i)
+        rAsInv(i, i) = 1.0 / rAs(i, i);
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
```

Comment:
could u describe what these functions are doing? (doxygen style)

Reply:
now explained and documented

Reply:
where? it is not here...